alibi_detect/cd/tensorflow/classifier.py

```diff
@@ -125,16 +125,21 @@ def __init__(
             retrain_from_scratch=retrain_from_scratch,
             seed=seed,
             input_shape=input_shape,
-            data_type=data_type
+            data_type=data_type,
         )
         if preds_type not in ['probs', 'logits']:
             raise ValueError("'preds_type' should be 'probs' or 'logits'")
 
         self.meta.update({'backend': Framework.TENSORFLOW.value})
 
         # define and compile classifier model
-        self.original_model = model
-        self.model = clone_model(model)
+        self.model: tf.keras.Model = model
```

Comment:
This change removes the public attribute `self.original_model`. This is still present in the scikit-learn and PyTorch backends. We might want to make `original_model` private in those detectors, although there are many things that are public but should be private, so leaving this for another PR. Below are some arguments for and against keeping a public `original_model` attribute.


- **Arguments for keeping**: The `model` provided to the detector will be trained during detector predictions. `detector.original_model` provides the user with access to their original model (if public attribute). If private, the original model might still be useful in the future e.g. if adding explanations functionality to the sklearn backend (based on discussion w/ @RobertSamoilescu, this was part of the reason for including `original_model` in the sklearn backend). 
- **Arguments for removing**: Unnecessary complexity. Would be good to remove all `model_clone`'s eventually (issues with subclassed models etc). Duplication of model (although weights need duplicating anyway...). If the user cares about the original model, they could just save (or clone) it before passing it to the detector. The attribute is currently undocumented so may as well be private...